protocol/src/test/scala/io/mediachain/protocol/DataObjectGenerators.scala

```diff
@@ -12,71 +13,134 @@ object DataObjectGenerators {
   import io.mediachain.util.cbor.CborAST._
   import io.mediachain.util.cbor.CValueGenerators._
 
+  val genStringMetas: List[Gen[(String, CValue)]] = (1 to 25).toList.map { _ =>
+    for {
+      key <- Gen.alphaStr
+      value <- genCPrimitive
+    } yield (key, value)
+  }
+
+  val genDateMeta: Gen[(String, CString)] = for {
+    date <- arbitrary[Date]
+  } yield ("date", CString(date.toString))
+
+  val genAuthorMeta: Gen[(String, CString)] = for {
+    author <- arbitrary[String] // TODO: make this a real special type
+  } yield ("author", CString(author))
 
   val genMeta: Gen[Map[String, CValue]] = for {
-    keys <- Gen.containerOf[List, String](arbitrary[String])
-    vals <- Gen.containerOfN[List, CValue](keys.length, genCPrimitive)
-    kvs = (keys, vals).zipped.toList
-  } yield Map[String, CValue](kvs:_*)
+    meta <- Gen.someOf[(String, CValue)](genAuthorMeta, genDateMeta, genStringMetas:_*)
+  } yield meta.toMap
 
   val genReference: Gen[Reference] = for {
     str <- arbitrary[String]
   } yield MultihashReference(
     MultiHash.hashWithSHA256(str.getBytes(StandardCharsets.UTF_8))
   )
 
-  val genEntity = for {
+  val genOptionalReference: Gen[Option[Reference]] =
+    Gen.option(genReference)
+
+  val genNilReference: Gen[Option[Reference]] = Gen.const(None)
+
+  val genEntity: Gen[Entity] = for {
     meta <- genMeta
   } yield Entity(meta)
 
-  val genArtefact = for {
+  val genArtefact: Gen[Artefact] = for {
     meta <- genMeta
   } yield Artefact(meta)
 
-  val genEntityChainCell = for {
-    entity <- genReference
-    chain <- genReference
+  def genReferenceFor(canonicalGen: Gen[CanonicalRecord]): Gen[Reference] =
+    for {
+      canonical <- canonicalGen
+    } yield MultihashReference.forDataObject(canonical)
+
+  def genEntityChainCell(
+    entityGen: Gen[Entity],
```

Comment:
if you want to fix a value you can use `Gen.const(someValue)`, which always returns the input.